FF1Lib/ItemMagic.cs

```diff
@@ -82,6 +82,16 @@ public void ShuffleItemMagic(MT19337 rng, Flags flags)
 				Spells = GetAllSpells(rng);
 			}
 
+			if ((bool)flags.StartingEquipmentRandomAoe && flags.ItemMagicMode != ItemMagicMode.None) {

```

Comment:
so that doesn't seem to do what is intended, the aoe spell is placed at the end of the list so it will never be picked, and because it doesn't discriminate it could introduce back excluded aoe spells (like nuke in balanced)

Reply:
That's a good point. Is there an existing "exclude spells" utility which could be used to filter the list before picking a new random spell?
And yes, I did think that we may be required to also remove a spell from the pool, good to know.

Reply:
as madmartin said you don't need to create a new list, the generated list has already all the valid spells, you need to find the aoe's in that list and move it up